### PR TITLE
NH-3970 - avoid losing original failure in case of additional failure

### PR DESCRIPTION
**Pull request by fredericDelaporte**
Test cases (explicit execution), and fix for [NH-3970](https://nhibernate.jira.com/browse/NH-3970) - TestCase base class: avoid hiding test failure on tear-down.

There is many things I would like to fix in the tests, but this one is really annoying. When a test fails but we cannot have its root cause (at least in NUnit gui), it is quite impractical.

### CODE REVIEW

File: src/NHibernate.Test/TestCase.cs [hunk 1]
```diff
@@ -154,30 +158,81 @@ protected virtual void OnTearDown()
...
+		{
+			var allClosed = true;
+			foreach (var session in _openedSessions)
 			{
-				log.Error("Test case didn't close a session, closing");
-				lastOpenedSession.Close();
-				return false;
+				if (session.IsOpen)
+				{
+					if (allClosed)
```
Comment (hazzik) [inline src/NHibernate.Test/TestCase.cs]: Maybe report each closing session separately? 

```
log.Error("Test case didn't close a session " + session.SessionId + " , closing");
```

Comment (fredericDelaporte) [inline src/NHibernate.Test/TestCase.cs]: And may help finding the session "at fault".

File: src/NHibernate.Test/TestCase.cs [hunk 2]
```diff
@@ -154,30 +158,78 @@ protected virtual void OnTearDown()
...
-			if (lastOpenedSession != null && lastOpenedSession.IsOpen)
+			var allClosed = true;
+			foreach (var session in _openedSessions)
 			{
-				log.Error("Test case didn't close a session, closing");
-				lastOpenedSession.Close();
-				return false;
+				if (session.IsOpen)
+				{
+					log.Error($"Test case didn't close session {(session as ISessionImplementor)?.SessionId}, closing");
```
Comment (hazzik) [inline src/NHibernate.Test/TestCase.cs]: It perhaps should be `session.GetSessionImplementor().SessionId`

Comment (fredericDelaporte) [inline src/NHibernate.Test/TestCase.cs]: Huh, I was not expecting the internal contract to be available from public interface!

**Pull request merged**